pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -84,6 +86,8 @@ public class SystemTopicBasedTopicPoliciesService implements TopicPoliciesServic
     @VisibleForTesting
     final Map<TopicName, List<TopicPolicyListener<TopicPolicies>>> listeners = new ConcurrentHashMap<>();
 
+    final Map<TopicName, BlockingDeque<CompletableFuture>> results = new ConcurrentHashMap<>();
```

Comment:
I don't think it will resolve the issue completely because the next request might happened on different brokers.

Reply:
All request for setting topic policy will redirect to the owner broker of the partition-0, so we do not need to consider distributed situations.

Reply:
we have such redirection logic in: `org.apache.pulsar.broker.admin.impl.PersistentTopicsBase#preValidation`
```
                        return getPartitionedTopicMetadataAsync(topicName, false, false)
                            .thenCompose(metadata -> {
                                if (metadata.partitions > 0) {
                                    return validateTopicOwnershipAsync(TopicName.get(topicName.toString()
                                    + PARTITIONED_TOPIC_SUFFIX + 0), authoritative);
                                } else {
                                    return validateTopicOwnershipAsync(topicName, authoritative);
                                }
                        });
```

Reply:
There are two other concerns about the fix:
- The problem still exists if the Topic Owner is transferred to another broker
- Now the reading is correct. But the topic may have yet to apply the latest policy even if users received the response.

Reply:
Indeed, even if all the read and update operations occur on the same broker, it remains crucial to serialize all such operations. Without serialization, the broker would still be incapable of ensuring that every modification is carried out on the most up-to-date data.

```
Update from v1 -> v2
Update from v1 -> v3
```

The changes from v2 will be missed.

Additionally, during the transfer of ownership, it is possible for ongoing updates to continue taking place on the old broker. Meanwhile, new requests directed to the new broker may still have an opportunity to update the data based on a potentially "dirty read."